### PR TITLE
Add --release-with-debug option to cargo-install-all.sh

### PR DESCRIPTION
**Pull request by willhickey**
#### Problem
Canaries use the release build, which limits debugging options

#### Summary of Changes
Add a --release-with-debug option that uses the release-with-debug profile. The canaries will use this via a corresponding change in cluster-ops scripts.

To see a sample run check out `mce1:~/build-artifacts/`

### CODE REVIEW

**Comment by steviez** [comment]
From your comment elsewhere, I think the only change we're making for building canaries is to use the different profile right? If so, do we still need the `--canary` option / maybe better to name the flag `release-with-debug` to match the profile ?

Also, I think the title and description are probably both a bit out of date now

**Comment by willhickey** [comment]
> From your comment elsewhere, I think the only change we're making for building canaries is to use the different profile right? If so, do we still need the `--canary` option / maybe better to name the flag `release-with-debug` to match the profile ?
> 
> Also, I think the title and description are probably both a bit out of date now

Updated. Once this is merged I'll run some tests of the other PR and backport this one to v1.17 and v1.16

**Review by steviez — APPROVED** [review]
LGTM assuming you tested locally

**Pull request merged**
